### PR TITLE
animating widget added.

### PR DESCRIPTION
**Pull request by AnubhavRawat02**
two targets are added. Widget extension and widget Intent extension. 
add more widget options by editing the PetActionsController class in Widget intent extension.

### CODE REVIEW

**Comment by curzel-it** [comment]
Thanks for the time you put in this!
There seems to be issues with iOS 17 and your code is not linted, so I cannot merge it.

With that said, I've tried everything that came to mind, but could not get widgets to update faster than 1 fps.
After checking lots of other apps, it seems that nobody is acually being able to break that barrier.

1 fps is great for a proof of concept, but I don't think I can deliver this feature unless I get up to 5 or 10.
Users would simply start review-bombing us!

TLDR; Thank you for your time, but I need to pass unfortunately!

**Comment by AnubhavRawat02** [comment]
Sorry to hear that you were not able to use my code.
At first I didn’t think it was even possible to make animations as they are
meant to give quick info to the users. But I learned a lot from this
project and i think the app you are working on is really cool!!

**Comment by curzel-it** [comment]
Thanks mate!
